rust/derive/Cargo.toml.in

```diff
@@ -0,0 +1,13 @@
+[package]
+name = "suricata-derive"
+version = "@PACKAGE_VERSION@"
+edition = "2018"
+
+[lib]
+proc-macro = true
+path = "@e_rustdir@/derive/src/lib.rs"
+
+[dependencies]
```

Comment:
are these build dependencies only? Or do they end up in the compiled bin?

Reply:
They are build only, but can't be labeled as [build-dependencies].  Within the scope of the derive crate, they are runtime dependencies, but only used by Suricata at a build time.

However, it doesn't appear the derive crate can be labeled as a build-dependency either, however, it does not end up in the output binary. Instead the derive crate is compiled and loaded by the compiler as a "compiler" plugin.  This becomes more visible in a cross compile scenario. The derive plugin would be compiled for the host environment, not the target environment.

Even though they already show up in our dependency tree, nothing triggers their usage yet so they will be new crates that we vendor.  They're rather ubiquitous for anyone building procedural macros.

Reply:
The last sentence on whether they will show up in the vendored crates is a bit unclear to me. Will they be included, but not end up in the compiled bin?

Reply:
They will be included in our vendor directory, compiled and run, but not in the output binary. However, their output (the generated code) will be in our output binary if that makes sense.

Much like the output of yacc, but where yacc is compiled and run as part of the build.